### PR TITLE
Mod/dev 4069 elasticsearch spending by cfda

### PR DESCRIPTION
**Pull request by BrianZito**
**Description:**

**Technical details:**
After looking at the legacy code, I decided it was better to keep cfda, naics, and psc as seperate classes

**Requirements for PR merge:**

1. [x] Unit & integration tests updated
2. [x] API documentation updated
3. [ ] Necessary PR reviewers:
    - [ ] Backend
    - [ ] Frontend 
4. [ ] Matview impact assessment completed
5. [ ] Frontend impact assessment completed
6. [ ] Data validation completed
7. [x] Appropriate Operations ticket(s) created (N/A)
8. [ ] Jira Ticket [DEV-4069](https://federal-spending-transparency.atlassian.net/browse/DEV-4069):
    - [x] Link to this Pull-Request
    - [ ] Performance evaluation of affected (API | Script | Download)
    - [ ] Before / After data comparison

**Area for explaining above N/A when needed:**
```
```

### CODE REVIEW

File: usaspending_api/api_docs/markdown/endpoints.md [hunk 1]
```diff
@@ -104,6 +104,7 @@ The currently available endpoints are listed in the following table.
 |[/api/v2/search/spending_by_category/awarding_subagency/](/api/v2/search/spending_by_category/awarding_subagency/)|POST| Returns data that is grouped in preset units to support the Spending by Awarding Subgency data visualizations on USAspending.gov's Advanced Search page|
 |[/api/v2/search/spending_by_category/funding_agency/](/api/v2/search/spending_by_category/funding_agency/)|POST| Returns data that is grouped in preset units to support the Spending by Funding Agency data visualizations on USAspending.gov's Advanced Search page|
 |[/api/v2/search/spending_by_category/funding_subagency/](/api/v2/search/spending_by_category/funding_subagency/)|POST| Returns data that is grouped in preset units to support the Spending by Funding Subgency data visualizations on USAspending.gov's Advanced Search page|
+|[/api/v2/search/spending_by_category/cfda/](/api/v2/search/spending_by_category/cfda/)|POST| Returns data that is grouped in preset units to support the Spending by CFDA Program data visualizations on USAspending.gov's Advanced Search page|
```
Comment (emily-brents) [inline usaspending_api/api_docs/markdown/endpoints.md]: As I recently found out, these are supposed to be in alphabetical order.

File: usaspending_api/search/v2/views/spending_by_category_views/spending_by_cfda.py [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+    "cfda_number": "code",
+    # Note: we could pull cfda title from the matviews but noticed the titles vary for the same cfda number
+    #       which leads to incorrect groupings
+    # 'cfda_title': 'name'
+}
+
+
+class CFDAView(BaseSpendingByCategoryViewSet, metaclass=ABCMeta):
+    """
+    Base class used by the different Awarding / Funding Agencies and Subagencies
```
Comment (kbard) [inline usaspending_api/search/v2/views/spending_by_category_views/spending_by_cfda.py]: ~Awarding / Funding Agencies and Subagencies~  CFDAs?

File: usaspending_api/api_contracts/contracts/v2/search/spending_by_category/cfda.md [hunk 3]
```diff
@@ -0,0 +1,166 @@
+FORMAT: 1A
+HOST: https://api.usaspending.gov
+
+# Spending By Funding Subagency [/api/v2/search/spending_by_category/cfda/]
```
Comment (emily-brents) [inline usaspending_api/api_contracts/contracts/v2/search/spending_by_category/cfda.md]: Should be "Spending By CFDA"

File: usaspending_api/api_contracts/contracts/v2/search/spending_by_category/cfda.md [hunk 4]
```diff
@@ -0,0 +1,166 @@
...
+
+This endpoint supports the advanced search page and allow for complex filtering for specific subsets of spending data.
+
+## POST
+
+This endpoint returns a list of the top results of Funding Subagencies sorted by the total amounts in descending order.
```
Comment (emily-brents) [inline usaspending_api/api_contracts/contracts/v2/search/spending_by_category/cfda.md]: also should be "CFDA" and not "Funding Subagencies

File: usaspending_api/api_contracts/contracts/v2/search/spending_by_category/cfda.md [hunk 5]
```diff
@@ -0,0 +1,166 @@
+FORMAT: 1A
+HOST: https://api.usaspending.gov
+
+# Spending By CFDA [/api/v2/search/spending_by_category/cfda/]
+
+This endpoint supports the advanced search page and allow for complex filtering for specific subsets of spending data.
+
+## POST
+
+This endpoint returns a list of the top results of CFDS codes sorted by the total amounts in descending order.
```
Comment (tony-sappe) [inline usaspending_api/api_contracts/contracts/v2/search/spending_by_category/cfda.md]:
```suggestion
This endpoint returns a list of the top results of CFDA codes sorted by the total amounts in descending order.
```

**Pull request closed without merging**